### PR TITLE
[validator] address edge case where nvidia-smi on host is empty

### PR DESCRIPTION
**Pull request by cdesiniotis**
This restores the logic that was accidentally removed in https://github.com/NVIDIA/gpu-operator/commit/c46ade78beebc4ae3dd7de934974f5871baa0bd3

### CODE REVIEW

**Review by tariq1890 — DISMISSED** [review]

File: validator/main.go [hunk 1]
```diff
@@ -693,6 +693,9 @@ func isDriverManagedByOperator(ctx context.Context) (bool, error) {
 
 func validateHostDriver(silent bool) error {
 	log.Info("Attempting to validate a pre-installed driver on the host")
+	if fileInfo, err := os.Lstat("/host/usr/bin/nvidia-smi"); err == nil && fileInfo.Size() == 0 {
```
Comment (tariq1890) [inline validator/main.go]: So if the file is not found, we should throw an error. This logic would be in an `else if` clause

Comment (cdesiniotis) [reply validator/main.go]: Updated.

File: validator/main.go [hunk 2]
```diff
@@ -693,6 +693,13 @@ func isDriverManagedByOperator(ctx context.Context) (bool, error) {
 
 func validateHostDriver(silent bool) error {
 	log.Info("Attempting to validate a pre-installed driver on the host")
+	fileInfo, err := os.Lstat("/host/usr/bin/nvidia-smi")
+	if err != nil {
+		return fmt.Errorf("no 'nvidia-smi' file present on the host")
```
Comment (tariq1890) [inline validator/main.go]: Let's log the error too.

Comment (cdesiniotis) [reply validator/main.go]: A separate log statement or append the error message to the error that gets returns here?

Comment (tariq1890) [reply validator/main.go]: the latter. Let's wrap the error

Comment (cdesiniotis) [reply validator/main.go]: Done.

**Review by tariq1890 — APPROVED** [review]

**Pull request merged**